### DISCUSSION TITLE
Expose API for getting and setting user status

### DISCUSSION THREAD

**Discussion by dylanstee-db** [community/community#42788] [Apps, API and Webhooks] [answered]
### Select Topic Area

General

### Body

It would be great to have an API for getting and setting a user's status (e.g., "On vacation"). This would allow companies to sync PTO/OOO status from Google Calendar/Outlook to GitHub to indicate when reviewers are unavailable (in GItHub parlance, "busy"). This could go into the existing [Get a user](https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28#get-a-user)/[Update the authenticated user](https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28#update-the-authenticated-user) APIs, or a new API dedicated to user status.

**Comment by petergoldsmith-anzx** [comment]
We find it very difficult to evaluate the effectiveness of the varying auto assignment algorithms (load balanced vs round robin) when we can't assess the 'availability' of the reviewers.

**Comment by Nyuwb** [comment]
I'd love to see this possibility in the GitHub API.

**Comment by caseyWebb** [comment]
Updating GitHub status is now the only part of our OOO workflow that is not automated via approval from our HR platform. As such (and considering it is not schedulable either), it's often forgotten/neglected.

**Comment by mattzhou-db** [comment]
+1 we really need this as well.

**Comment by Xatenev** [comment]
I may be missing something, but it does seem to be possible to retrieve the user status, for example to retrieve whether the user is busy, by using the GraphQL API.

```
query {
  user(login: "username") {
    status {
      emoji
      message
      indicatesLimitedAvailability # This field indicates if the user is busy
    }
  }
}
```

For example:

```
curl -H "Authorization: bearer YOUR_ACCESS_TOKEN" -X POST -d '{"query": "query { user(login: \"username\") { status { emoji message indicatesLimitedAvailability } } }"}' https://api.github.com/graphql
```

**Comment by dylanstee-db** [comment]
@Xatenev, do you have any links to the docs for this endpoint? I haven't used the GQL API for Github before. Is there a similar `mutation` to update the `indicatesLimitedAvailability`?

**Reply by Xatenev** [reply]
@dylanstee-db See https://docs.github.com/en/graphql/reference/objects#userstatus for the documentation. I think that updating it via the GraphQL API is currently not possible. It seems to be more focussed on querying data rather than updating.

**Reply by dylanstee-db** [reply]
Thanks for the reference. Querying is certainly useful, but I need the ability to set status as well. Hopefully Github will add that functionality soon.

**Comment by fadedlamp42** [comment]
We use [Flamingo](flamingoapp.com/) for our OOO tracking and reached out to them for an integration, but seems like this API limitation is preventing them (and us) from building this.

**Comment by blipinsk** [comment, marked as answer]
There is [`changeUserStatus` mutation](https://docs.github.com/en/graphql/reference/mutations#changeuserstatus) which accepts [`ChangeUserStatusInput`](https://docs.github.com/en/graphql/reference/input-objects#changeuserstatusinput) containing `limitedAvailability` `Boolean` field.



|  `changeUserStatus` | `ChangeUserStatusInput` |
|--------|-------|
| <img src="https://github.com/user-attachments/assets/dfb40cb8-568c-46fa-832f-b4ef55c872d8" width="500" /> | <img src="https://github.com/user-attachments/assets/6e5417e4-76eb-4b53-a03f-39dd5c86f68b" width="500" /> |
